python/pyiceberg/cli/console.py

```diff
@@ -0,0 +1,264 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+import os
+from typing import Literal, Optional, Tuple
+
+import click
+from rich.console import Console
+from rich.table import Table
+from rich.tree import Tree
+
+from pyiceberg.catalog import Catalog
+from pyiceberg.catalog.hive import HiveCatalog
+from pyiceberg.catalog.rest import RestCatalog
+from pyiceberg.cli.output import ConsoleOutput, JsonOutput, Output
+from pyiceberg.exceptions import NoSuchNamespaceError, NoSuchTableError
+
+
+@click.group()
+@click.option("--catalog", default="catalog")
+@click.option("--output", type=click.Choice(["text", "json"]), default="text")
+@click.option("--uri", default=lambda: os.environ.get("PYICEBERG_URI"))
+@click.option("--credential", default=lambda: os.environ.get("PYICEBERG_CREDENTIAL"))
+@click.pass_context
+def run(ctx, catalog: str, output: str, uri: str, credential: str):
+    console = Console()
+    console.print("Welcome to PyIceberg 🐧")
```

Comment:
Cute, but should we print a welcome message every execution?

Reply:
Probably we want to get rid of this :p